dask/array/tests/test_linalg.py

```diff
@@ -41,6 +41,49 @@ def test_tsqr_irregular_blocks():
     assert_eq(r, np.triu(r))  # r must be upper triangular
 
 
+def test_sfqr_regular_blocks():
+    m, n = 10, 40
+    mat = np.random.rand(m, n)
+    data = da.from_array(mat, chunks=(10, 10), name='A')
+
+    q, r = sfqr(data)
+    q = np.array(q)
+    r = np.array(r)
```

Comment:
You don't need to explicitly compute `q` and `r` here.  Instead it's best to leave them as dask arrays and pass those into assert_eq, which will both call compute appropriately and run a number of other checks on the dask objects.

Reply:
You might also consider testing the following:

```python
q, r = da.linalg.qr(data)
qq, rr = np.linalg.qr(mat)

assert_eq(q, qq)
assert_eq(r, rr)
```

This would, I think, check a number of concerns directly

Reply:
It would also be good to verify that this works under a few chunking situations, notably where there is only one chunk, and when chunks are uneven.

Reply:
Ah, I see that you handle exactly the chunking cases that I mention here below.

If it makes sense you might consider using `pytest.mark.parameterize` to combine a few tests.  Here is a naive example that might not be appropriate for your situation.

```python
@pytest.mark.parameterize('chunks', [(10, 10), (10, 7), (10, 40)])
def test_sfqr(chunks):
    mat = ...
    data = da.from_array(mat, chunks=chunks)

```

Reply:
I see:
```
q = np.array(q)
r = np.array(r)
```
is not necessary. I'll change the existing tsqr tests too.